tags/key.go

```diff
@@ -15,49 +15,94 @@
 
 package tags
 
-import "fmt"
-
-var keys []Key
+// Modifier functions modifies a tag set.
+type Modifier func(t *TagSet) *TagSet
```

Comment:
What about:

```go
type Modifer struct {
  f func(t *TagSet) *TagSet
}
```

I'd like to prevent a user from write their own function to modify a `TagSet`, since it seems like `TagSet`s are intended to be immutable now.

Bikeshedding: I think I've seen the term "mutator" used to describe this kind of thing.  Doesn't really matter much to me, but I thought I'd mention.

Reply:
Done

Reply:
(See also my justification against interfaces with unexported methods over on #41.)